pkg/controllers/leaderworkerset_controller.go

```diff
@@ -394,7 +406,18 @@ func (r *LeaderWorkerSetReconciler) updateConditions(ctx context.Context, lws *l
 		lws.Status.UpdatedReplicas = int32(updatedCount)
 		updateStatus = true
 	}
-	condition := makeCondition(updatedAndReadyCount == int(*lws.Spec.Replicas))
+
+	condition := makeCondition(leaderworkerset.LeaderWorkerSetProgressing)
+
+	if rollingUpdateCount < currentWorkerCount && lws.Spec.RolloutStrategy.Type == "RollingUpdate" {
+		setCondition(lws, condition)
+		condition = makeCondition(leaderworkerset.LeaderWorkerSetUpgradeInProgress)
+	}
+
+	if updatedAndReadyCount == int(*lws.Spec.Replicas) {
+		condition = makeCondition(leaderworkerset.LeaderWorkerSetAvailable)
+	}
+
 	updateCondition := setCondition(lws, condition)
```

Comment:
The main reason why I call `setCondition` twice is because `Progressing` condition has to be set if `UpgradeInProgress` is set. Instead I could add an special condition in `setCondition` to set both `Progressing` and `UpgradeInProgress` if `UpgradeInProgress` is pass to the function, but calling it twice is more explicit. @liurupeng thoughts?

Reply:
refactored